### PR TITLE
[ROS-435] Port existing RblString primitives to current code base and refactor fixnum/rblfloat

### PR DESCRIPTION
**Pull request by AbnerZheng**
## Overview
Port existing RblString primitives to current code base and refactor fixnum/rblfloat

### Does this PR relate to an RChain JIRA issue? 
https://rchain.atlassian.net/browse/ROS-435

### Complete this checklist before you submit the PR
- [ ] This PR contains no more than 200 lines of code, excluding test code.
- [x] This PR meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards).
- [x] You have someone in mind to assign for review. Make this assignment after submitting the PR.

### CODE REVIEW

**Comment by AbnerZheng** [comment]
@tymm Please take a look.

File: roscala/src/main/scala/coop/rchain/roscala/prim/rblfloat.scala [hunk 1]
```diff
@@ -33,16 +31,16 @@ object rblfloat {
     override val maxArgs: Int = 2
 
     @checkTypeMismatch[RblFloat]
-    override def fn(ctxt: Ctxt, globalEnv: GlobalEnv): Ob =
+    override def fnSimple(ctxt: Ctxt): Either[PrimError, RblFloat] =
       ctxt.nargs match {
         case 1 =>
           val n = ctxt.argvec.value.head.asInstanceOf[RblFloat]
-          -n
+          Right(RblFloat(-n.value))
```
Comment (tymm) [inline roscala/src/main/scala/coop/rchain/roscala/prim/rblfloat.scala]: Why not `Right(-n)`?

Comment (AbnerZheng) [reply roscala/src/main/scala/coop/rchain/roscala/prim/rblfloat.scala]: done

**Comment by AbnerZheng** [comment]
it is so stupid!!!

**Pull request closed without merging**